### PR TITLE
Add scrolling to proposals

### PR DESCRIPTION
**Pull request by alhonaut**

### CODE REVIEW

File: src/app/components/job-categories.tsx [hunk 1]
```diff
@@ -1,12 +1,40 @@
...
-const JobCategories: NextPage<JobCategoriesType> = ({ className = "" }) => {
+const PROPOSALS_TO_FETCH = 5; // Amount of proposals that would be loaded after scroll
+
+const JobCategories: NextPage<JobCategoriesType> = ({ initialData, className = "" }) => {
+  const [gigData, setGigData] = useState<any[]>(initialData); // further replace any[] to Gig[] Type from @types
+  const [offset, setOffset] = useState(PROPOSALS_TO_FETCH);
+  const [hasMore, setHasMore] = useState(true);
+  const [scrollTrigger, isView] = useInView();
+
+  const loadMoreProposals = async () => {
```
Comment (mgroovyank) [inline src/app/components/job-categories.tsx:22]: please add comments defining what the function does

**Review by mgroovyank — CHANGES_REQUESTED** [review]

File: src/app/components/job-categories.tsx (hunk 1 above)
Comment (alhonaut) [reply src/app/components/job-categories.tsx:22]: First hook check if we run out with proposals from supabase
Second check if user scroll down page to load more proposals to frontend

File: src/app/actions/get-proposals.ts [hunk 2]
```diff
@@ -0,0 +1,15 @@
+"use server";
+import { supabase } from "@/src/utils/supabase";
+
+export const getActiveProposals = async (currPage: number, limitPage: number) => {
```
[the remark this answers is not in the document]
Comment (mgroovyank) [reply src/app/actions/get-proposals.ts:4]: refer https://github.com/GiG2Hire/gig-protocol/blob/main/src/app/actions/create-gig.ts

**Comment by mgroovyank** [comment]
From next time, don't use test as branch names. use something, like feature/load-proposals

**Pull request closed without merging**